src/search/pdbs/cegar.cc

```diff
@@ -385,7 +389,7 @@ bool CEGAR::get_flaws_for_pattern(
                 log << "plan did not lead to a goal state: ";
             }
             bool raise_goal_flaw = false;
-            for (const FactPair &goal : goals) {
+            for (const FactPair &goal : use_restricted_goal ? goals : task_properties::get_fact_pairs(task_proxy.get_goals())) {
```

Comment:
I find this a bit unintuitive. At this point `goals` is something that is passed to the class in the constructor. Instead of passing `goals` and an option that essential is `ignore_the_goal_I_gave_you_and_use_something_else`, can we handle the option earlier and just pass a modified goal vector to the class?